web/app/test/ServiceMeshTest.jsx

```diff
@@ -38,6 +39,32 @@ describe('ServiceMesh', () => {
     });
   });
 
+  it("renders the spinner before metrics are loaded", () => {
+    fetchStub.returnsPromise().resolves({
+      ok: true,
+      json: () => Promise.resolve({ metrics: [] })
+    });
+    component = mount(routerWrap(ServiceMesh));
+
+    expect(component.find("ConduitSpinner")).to.have.length(1);
+    expect(component.find("ServiceMesh")).to.have.length(1);
+    expect(component.find("CallToAction")).to.have.length(0);
+  });
+
+  it("renders a call to action if no metrics are received", () => {
+    fetchStub.returnsPromise().resolves({
+      ok: true,
+      json: () => Promise.resolve({ metrics: [] })
+    });
+    component = mount(routerWrap(ServiceMesh));
+
+    return withPromise(() => {
+      expect(component.find("ServiceMesh")).to.have.length(1);
+      expect(component.find("ConduitSpinner")).to.have.length(0);
+      expect(component.find("CallToAction")).to.have.length(1);
+    });
+  });
+
   it("displays message for no deployments detetcted", () => {
```

Comment:
since there are a lot of these "displays message for...." tests, it might be nice to wrap them in a Describe block, and then reword the tests to be clearer about what behaviours we're testing. In the `it()` assertions, I think it's usually clearer to describe the specific behaviours we're testing e.g. `it("displays a message when there are no deployments", ...)` but in the describe, since we're grouping these, we can write the method name e.g.
`describe("renderAddDeploymentsMessage", ...)`

In general, we should try to think about wording the tests to the specific behaviours we're testing, and try to be as descriptive as possible (which sometimes means trying to say /what/ you're testing without repeating what you're asserting in the test). E.g. with these tests, we're testing that the correct message displays in certain conditions. So we can try to reflect that in the wording of the assertion, such that the tests read something like:
```
 ServiceMesh
    renderAddDeploymentsMessage
       displays when no deployments are in the mesh
```

Reply:
ah yes, that's awesome. I didn't know you can nest describes, but it will definitely make it clearer. thanks!